File: Python/1734. 员工的重要性.py
# If you need to import additional packages or classes, please import here.

def func():
    
    # please define the python3 input here. For example: a,b = map(int, input().strip().split())
    
    # please finish the function body here. 
    
    # please define the python3 output here. For example: print().
    
    while True:
        try:
            ss = input().strip()
            ss = ss[2:-2].split('], [')
            n = input()
            importance = [0]
            relationship = [[]]
            for s in ss:
                before, sup = s.split(', [')
                index, weight = before.split(', ')
                importance.append(weight)
                relationship.append(list(sup[:-1].split(', ')))
            result = 0
            queue = [n]
            
            while queue:
                x = queue.pop(0)
                if x == '':
                    continue
                result += int(importance[int(x)]) 
                queue += relationship[int(x)]
            
            print(result)
            
        except EOFError:
            break

if __name__ == "__main__":
    func()

File: C++/1734. 员工的重要性.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include <map>
#include <vector>
#include <queue>
int main()
{

    // please define the C++ input here. For example: int a,b; cin>>a>>b;;
    
    // please finish the function body here.
    
    // please define the C++ output here. For example:cout<<____<<endl;
    
    string ss;
    int n;
    
    while(getline(cin,ss)) {
        cin>>n;
        vector<int> importance;
        importance.push_back(0);
        map<int, vector<int>> relationship;
        int flag = 0;
        int id = 0;
        int w = 0;
        int temp = 0;
        vector<int> arr;
        
        ss = ss.substr(1, ss.size() - 2);
        
        for(auto c:ss) {
            if(c == ' ') {
                continue;
            }
            if(c == '[') {
                flag++;
            } else if(isdigit(c)) {
                temp = temp * 10 + (c - '0');
            } else if(c == ',') {
                if(temp == 0) {
                    continue;
                }
                if(flag == 1) {
                    if(id == 0) {
                        id = temp;
                        temp = 0;
                    } else {
                        w = temp;
                        temp = 0;
                        importance.push_back(w);
                    }
                } else {
                    arr.push_back(temp);
                    temp = 0;
                }
            } else {
                flag--;
                if(id != 0) {
                    relationship[id] = arr;
                    id = 0;
                    arr = {};
                }
            }
        }
        
        int result = 0;
        queue<int> emp;
        emp.push(n);
        
        while(!emp.empty()) {
            int x = emp.front();
            emp.pop();
            result += importance[x];
            for(auto i:relationship[x]) {
                emp.push(i);
            }
        }
        cout<<result<<endl;
    }

return 0;
}
